aten/src/ATen/native/Itertools.cpp

```diff
@@ -46,7 +46,10 @@ Tensor cartesian_prod(TensorList tensors) {
 
 Tensor combinations(const Tensor& self, int64_t r, bool with_replacement) {
   TORCH_CHECK(self.dim() == 1, "Expect a 1D vector, but got shape ", self.sizes());
-  TORCH_CHECK(r > 0, "Expect a positive number, but got ", r);
+  TORCH_CHECK(r >= 0, "Expect a non-negative number, but got ", r);
```

Comment:
Hey @johnpeterson123, thanks for the fix.

While we're here let's fix up a couple things. First, let's change the start of each `TORCH_CHECK` string to "torch.combinations expected ..." so users know where the error message is originating from. Second would you add a comment describing the `r == 0` short-circuit?

Reply:
Finally, there's an OpInfo for combinations in test_common_methods.py which has a lot of metadata for how the operation works. See 

https://github.com/pytorch/pytorch/blob/bf610f08b07a1135eaffd1ee53a1b4aa901601c1/torch/testing/_internal/common_methods_invocations.py#L8757

Would you add two ErrorInputs to this OpInfo, one for `r < 0` and another for `dim != 1` following the pattern used by `torch.kthvalue`?

https://github.com/pytorch/pytorch/blob/bf610f08b07a1135eaffd1ee53a1b4aa901601c1/torch/testing/_internal/common_methods_invocations.py#L9711

If that's confusing then no worries, we can always add those error cases later.